### PR TITLE
Smoke Test: use configurable authority host

### PR DESCRIPTION
**Pull request by danieljurek**
use EnvironmentCredentialBuilder to build client with configurable authority host

### CODE REVIEW

File: common/smoke-tests/pom.xml [hunk 1]
```diff
@@ -90,7 +90,7 @@
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-core</artifactId>
-      <version>1.2.0</version>
+      <version>1.4.0-beta.1.dev.20200311.1</version>
```
Comment (JimSuplizio) [inline common/smoke-tests/pom.xml]: This probably shouldn't be being set to a dev version and checked in. Maybe updating the dependencies to the latest GA version would be better?

Comment (danieljurek) [reply common/smoke-tests/pom.xml]: Definitely shouldn't have dev versions in there. I'll revert that.

File: common/smoke-tests/src/main/java/com/azure/KeyVaultSecrets.java [hunk 2]
```diff
@@ -47,16 +47,19 @@ public static void main(String[] args) {
         LOGGER.info("IDENTITY - CREDENTIAL");
         LOGGER.info("---------------------");
 
-        /* DefaultAzureCredentialBuilder() is expecting the following environment variables:
+        /* EnvironmentCredentialBuilder() is expecting the following environment variables:
```
Comment (jongio) [inline common/smoke-tests/src/main/java/com/azure/KeyVaultSecrets.java]: I would prefer that we keep the test with DAC and then use reflection to set the authority host until this is fixed:  https://github.com/Azure/azure-sdk-for-java/issues/8626

That way the test is consistent across languages: Test DAC and we can remove the reflection once that issue is resolved.  @g2vinay

Comment (jongio) [reply common/smoke-tests/src/main/java/com/azure/KeyVaultSecrets.java]: It was implemented in the beta here: https://github.com/Azure/azure-sdk-for-java/blob/5dd0c40303aaf85aac4e7cf36b64a0ab1fe62437/sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java#L21

So, ideally this test runs against beta and we use that.

Comment (danieljurek) [reply common/smoke-tests/src/main/java/com/azure/KeyVaultSecrets.java]: Smoke tests run from the latest nightly packages available. Here it is succeeding with the new syntax: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=296102&view=results

**Comment by danieljurek** [comment]
> Cosmos released [4.0.1-beta.1](https://search.maven.org/search?q=g:com.azure%20AND%20a:azure-cosmos) on Monday from master. You should be able to uncomment the dependency in the pom file.

@JimSuplizio -- I've opened https://github.com/Azure/azure-sdk-for-java/issues/9086 to track this. It's outside the scope of this change.

**Pull request merged**